src/components/MAPI/organizations/OrganizationDetailGeneral/utils.ts

```diff
@@ -56,51 +69,59 @@ export function fetchClustersSummaryKey(
 async function fetchSingleClusterSummary(
   httpClientFactory: HttpClientFactory,
   auth: IOAuth2Provider,
-  cluster: capiv1alpha3.ICluster
+  cluster: capiv1alpha3.ICluster,
+  CPNodesPermissions: IPermissions,
+  nodePoolsPermissions: IPermissions
 ): Promise<ui.IOrganizationDetailClustersSummary> {
   const summary: ui.IOrganizationDetailClustersSummary = {};
 
   const machineTypes = getMachineTypes();
 
-  try {
-    const cpNodes = await fetchControlPlaneNodesForCluster(
-      httpClientFactory,
-      auth,
-      cluster
-    );
-
-    appendControlPlaneNodeStats(cpNodes, machineTypes, summary);
-  } catch (err) {
-    ErrorReporter.getInstance().notify(err as Error);
+  if (CPNodesPermissions.canList) {
+    try {
+      const cpNodes = await fetchControlPlaneNodesForCluster(
```

Comment:
Since the function for fetching cluster summary includes fetching several different resources, we pass in the appropriate permissions as arguments and only fetch the resources the user has permissions for.